classes/Manufacturer.php

```diff
@@ -381,7 +381,7 @@ public static function getProducts(
         }
 
         $front = true;
-        if (!in_array($context->controller->controller_type, ['front', 'modulefront'])) {
+        if (isset($context->controller) && !in_array($context->controller->controller_type, ['front', 'modulefront'])) {
```

Comment:
```suggestion
        if (!isset($context->controller) || !in_array($context->controller->controller_type, ['front', 'modulefront'])) {
```
If the controller is not accessible it is very likely because it's a context that is not the front no?

Reply:
We could also have a context in the admin, couldn't we? I think we should only serve back data if we really need it, so we must provide a proper context in this case.

Thanks for the review ! :-)

Reply:
Yes but the current implementation tends to fallback more easily on the default `$front = true;` But the absence of a controller to check is more likely to happen in a context where we are NOT on the front

The Admin API is a good example since it is what motivated this fix, I may be wrong but with the current fix in admin API where we don't have a legacy controller accessible the Manufacturer will "think" it is in FO

That's why I suggested the inverse condition when no controller is detected we should have `$front = false;` hence the above suggestion

Reply:
I understand your point of view and have applied your suggestion.

However, I think it would be beneficial to have some kind of "admin" controller when using the API. I feel there is a potential security issue here, as we automatically serve back data when no controller is set.

Thanks!

Reply:
I tested this change and it works correctly:
<img width="1903" height="5399" alt="Screenshot 2025-11-21 at 08-53-34 " src="https://github.com/user-attachments/assets/2ad62d45-924a-401b-9a88-04c295768cac" />